pkg/infrastructure/services/deployment.go

```diff
@@ -26,9 +26,18 @@ const (
 	portName      = "http"
 	labelAppKey   = "app"
 	singleReplica = int32(1)
+
+	// AppPropContentHashKey is the annotation key for the content hash of application.properties
+	AppPropContentHashKey = "appPropContentHash"
+	// AppPropVolumeName is the name of the volume for application.properties
+	AppPropVolumeName = "app-prop-config"
+
+	appPropDefaultMode = int32(420)
+	appPropFileName    = "application.properties"
+	appPropFilePath    = "/home/kogito/config"
 )
 
-func createRequiredDeployment(service v1alpha1.KogitoService, resolvedImage string, definition ServiceDefinition) *appsv1.Deployment {
+func createRequiredDeployment(service v1alpha1.KogitoService, resolvedImage string, appPropContentHash string, definition ServiceDefinition) *appsv1.Deployment {
```

Comment:
Please, instead of adding the configMap here, I'd say to leave it to the `resources.go` file to do it. This function supposed to return only the **raw** `Deployment` object and then [we use other functions to decorate it](https://github.com/kiegroup/kogito-cloud-operator/blob/master/pkg/infrastructure/services/resources.go#L88), as we are doing when injecting Kafka or Infinispan variables. The support functions could go into `configmap.go` or in the decorator itself.